wallet-new/src/Cardano/Wallet/Kernel/Keystore.hs

```diff
@@ -0,0 +1,239 @@
+{-- | An opaque handle to a keystore, used to read and write 'EncryptedSecretKey'
+      from/to disk.
+
+    NOTE: This module aims to provide a stable interface with a concrete
+    implementation concealed by the user of this module. The internal operations
+    are currently quite inefficient, as they have to work around the legacy
+    'UserSecret' storage.
+--}
+
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+
+module Cardano.Wallet.Kernel.Keystore (
+      Keystore -- opaque
+      -- * Constructing a keystore
+    , bracketKeystore
+    , bracketLegacyKeystore
+    -- * Destroying a keystore (something you rarely should do)
+    , releaseAndDestroyKeystore
+    -- * Releasing a keystore and its associated resources
+    , releaseKeystore
```

Comment:
We should not have `releaseKeystore` exported anymore (it breaks the invariants established by the brackets).